Refresh the AWS app access guide

ptgott opened the pull request:
Closes #36458 (tracking issue)

Other issues closed:
--------------

Add a statement of benefits, a brief architecture overview, more troubleshooting information, and a summary of IAM resources created in this guide.

Closes #10853

Add troubleshooting re: trust relationships and
checking logs.

Closes #17880

Add more explicit trust relationship guidance and uses CLI examples instead of console screenshots.

Closes #34167

Add more context re: the trust relationship
instructions.

Closes #11797

Add instructions that a user can follow to deploy the Application Service on EKS.

Closes #24373

Other changes to make the guide easier to follow:
-------------------------------------------------

- Only show one IAM role. Remove the `ExamplePowerUser` role for simplicity.
- Use the `aws` CLI for creating IAM resources. This is faster to follow than the current approach, takes up less page real estate, and doesn't require screenshots. Using the CLI  also makes the trust relationship instructions more explicit.
- Move external links to a "Further reading" section so readers can get through the guide more quickly.
- Remove the video, since it's long out of date.
- Reduce the number of steps: Convert steps into H3s in order to make it easier to form a mental model of the guide and make the step count less intimidating.
- Add EC2 instance profile instructions.
- Add a quick boto3 python example for `tsh proxy aws`

ptgott committed:
Refresh the AWS app access guide

Issues closed:
--------------

Add a statement of benefits, a brief architecture overview, more
troubleshooting information, and a summary of IAM resources created in
this guide.

Closes #10853

Add troubleshooting re: trust relationships and
checking logs.

Closes #17880

Add more explicit trust relationship guidance and uses
CLI examples instead of console screenshots.

Closes #34167

Add more context re: the trust relationship
instructions.

Closes #11797

Add instructions that a user can follow to deploy the
Application Service on EKS.

Closes #24373

Other changes to make the guide easier to follow:
-------------------------------------------------

- Only show one IAM role. Remove the `ExamplePowerUser` role for
  simplicity.
- Use the `aws` CLI for creating IAM resources. This is faster to follow
  than the current approach, takes up less page real estate, and doesn't
  require screenshots. Using the CLI  also makes the trust relationship
  instructions more explicit.
- Move external links to a "Further reading" section so readers can get
  through the guide more quickly.
- Remove the video, since it's long out of date.
- Reduce the number of steps: Convert steps into H3s in order to make it
  easier to form a mental model of the guide and make the step count
  less intimidating.
- Add EC2 instance profile instructions.
- Add a quick boto3 python example for `tsh proxy aws`

greedy52 reviewed (commented):
Going forward, do we prefer to use command line tools like `aws`, `az` for configuring the cloud or it depends on the content? 

For example, I am working on GCP MySQL IAM auth. The current guide is all screenshots.

ptgott committed:
Respond to greedy52 feedback

- Say "EKS" instead of just "Kubernetes" in the prerequisites.
- Edit the trust policy for the `ExampleReadOnlyAccess` role in the
  cross-account use case to include an external ID and require a
  specific role for the principal.

ptgott commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
```diff
@@ -1,186 +1,390 @@
+   </TabItem>
+   <TabItem label="EKS">
+
+   Retrieve your OIDC issuer ID, assigning <Var name="cluster-name" /> to the name
+   of your EKS cluster and <Var name="eks-region" /> to the AWS region where your
+   EKS cluster is running:
```
Should we recommend running `eksctl create iamserviceaccount` with the `--override-existing-serviceaccounts` flag and a `--name` flag pointing to the `teleport-kube-agent` service account? 

One consideration is that, in this guide, we tell the user to install the `teleport-kube-agent` Helm chart (which creates a service account for the Teleport Application Service) after creating the IAM role.

greedy52 replied:
> One consideration is that, in this guide, we tell the user to install the teleport-kube-agent Helm chart (which creates a service account for the Teleport Application Service) after creating the IAM role.

I see. If that's the case then it's good. 

To be honest I don't know what ppl do for IAM role for service account in the field. Personally I would still use eksctl since it's easier, then use that created service account to create App service with Helm.

Maybe @hugoShaka @anto have more insight on this.

ptgott commented:
@kimlisa @mdwn would one of you have time to review this one? Thanks!

kimlisa commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
```diff
@@ -1,186 +1,395 @@
 
-![Select Role Step 2](../../../img/application-access/create-role-example-readonly-2.png)
+1. Create a file called `aws-ro-access.yaml` with the following content,
+   replacing <Var name="AWS_ACCOUNT" /> with the account number where you
```
we refer it as ID elswhere
```suggestion
   replacing <Var name="AWS_ACCOUNT" /> with the account ID where you
```

kimlisa commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
````diff
@@ -190,39 +399,88 @@ Application Service:
+       uri: "https://console.aws.amazon.com/ec2/v2/home"
+   ```
+
+1. Edit `/etc/teleport.yaml` to replace `teleport.example.com:443` with the host
````
can we use the same `in place editor` thingy we do elsewhere in this guide for writing over `teleport.example.com:443`?

kimlisa commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
```diff
@@ -190,39 +399,88 @@ Application Service:
+<TabItem label="EKS">
+
+1. Retrieve the join token you created earlier in this guide by running the
+   following command and copying a token with the `App` type:
```
```suggestion
   following command and copy the token with the `App` type:
```

kimlisa commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
```diff
@@ -269,44 +527,136 @@ AWS roles in these accounts.
+   ![IAM role selector](../../../img/application-access/iam-role-selector.png)
 
-![IAM role selector](../../../img/application-access/iam-role-selector.png)
+1. Click on the role you want to assume and you will get redirected to the AWS
```
didn't i already click on the role on the previous step? 
```suggestion
1. You will get redirected to the AWS
```

kimlisa commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
```diff
@@ -269,44 +527,136 @@ AWS roles in these accounts.
+   into the search box if there are too many resources to view all of them at
+   once.)
+
+1. Click the **Launch** button for the AWS Console application, then choose the
```
```suggestion
1. Click the **Launch** button for the AWS Console application, then click on the
```

kimlisa commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
```diff
@@ -269,44 +527,136 @@ AWS roles in these accounts.
+### Access the AWS CLI
 
-![Federated login](../../../img/application-access/federated-login@2x.png)
+1. Log into the previously configured AWS console app on your desktop:
```
is the `previously` necessary? (b/c i didn't know what was previously configured)
```suggestion
1. Log into the configured AWS console app on your desktop:
```

kimlisa commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
```diff
@@ -269,44 +527,136 @@ AWS roles in these accounts.
+
+### Access applications using AWS SDKs
+
+1. Log into the AWS Management Console app you configured earlier:
```
hrmm, i think either remove the `configured earlier` or perhaps (we can also probably word it similarly in the comment above this one)
```suggestion
1. Log into the AWS Management Console app you configured from this guide:
```

kimlisa commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
```diff
@@ -269,44 +527,136 @@ AWS roles in these accounts.
+
+1. Use the following command to start a local HTTPS proxy server, which forwards
+   AWS API traffic to the Teleport Application Service. Leave the terminal where
+   you run the command open since it runs in the foreground:
```
```suggestion
   you ran the command open since it runs in the foreground:
```

ptgott replied:
Suggested a clearer wording since "run" and "ran" can be ambiguous in this paragraph.

kimlisa commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
```diff
@@ -316,85 +666,61 @@ username which you can search for to get the events history:
 
-Now, use the following command to start a local HTTPS proxy server your
-applications will be connecting to:
+In the Teleport Application Service fails to assume the `ExampleReadOnlyAccess`
```
```suggestion
If the Teleport Application Service fails to assume the `ExampleReadOnlyAccess`
```

kimlisa commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
```diff
@@ -445,5 +771,46 @@ instance and attach an IAM role to it.
 
-- Take a closer look at [role-based access controls](../controls.mdx).
-- Explore other application access [guides](../guides.mdx).
+Now that you know how set up Teleport to protect access to the AWS Management
```
```suggestion
Now that you know how to set up Teleport to protect access to the AWS Management
```

ptgott commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
```diff
@@ -1,186 +1,395 @@
+
+### Configure a role to enable access to
+
+In this section, you will configure an AWS role to enable access to with the
+Teleport Application Service.
```
I agree that this sentence is unclear, but I'm not sure this rephrasing helps, since it sounds like only the Application Service requires this access. I've suggested adding more context to this paragraph.

ptgott committed:
Respond to kimlisa feedback

- Fix unclear subsection intro sentence.
- Fix a potentially confusing reference to Step 1.
- Fix an unclear use of "configuration".
- Clarify a heading.
- Minor clarity/accuracy improvements.
- Use a Var component for the Proxy Service address in the Linux server
  config file snippet.

ptgott commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
```diff
@@ -1,186 +1,395 @@
+|`AssumeRole`|IAM policy|Allows the Application Service to assume other roles in order to proxy user requests to AWS.|
+|`TeleportAWSAccess` (for EC2 deployments) |EC2 instance profile|Associates the `TeleportAWSAccess` role with your EC2 instance.|
+
+### Configure a role to enable access to
```
I've suggested "Configure a role for Teleport users to request", which I think makes more sense for this setup step. Otherwise, this section seems to overlap with the later section called, "Give the Teleport Application Service permissions to assume roles".

ptgott commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
```diff
@@ -1,186 +1,395 @@
+   In the configuration we show in this guide, the Teleport Application Service
+   assumes the `TeleportAWSAccess` role, then uses that role to assume the
+   `ExampleReadOnlyAccess` role. With the trust policy above, AWS authorizes
+   this operation. (We will create the `TeleportAWSAccess` role later in Step
+   1.)
```
The role creation happens in the section, "Give the Teleport Application Service permissions to assume roles", which is later in step 1. If this is causing confusion, though, I'll change the wording to "later in this guide".

ptgott commented on docs/pages/application-access/cloud-apis/aws-console.mdx:
````diff
@@ -1,186 +1,395 @@
+   }
+   ```
+
+   In the configuration we show in this guide, the Teleport Application Service
````
"Configuration" should be "setup", referring to everything the reader will have accomplished by the time they finish the guide.
